compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -1264,6 +1264,7 @@ void quantize_const_inputs(luci::CircleNode *node, loco::DataType output_type)
     case luci::CircleOpcode::EXP:
     case luci::CircleOpcode::FLOOR:
     case luci::CircleOpcode::FLOOR_DIV:
+    case luci::CircleOpcode::GATHER:
```

Comment:
If `Gather` is added to this section, all the inputs are used to quantize AFAIK.

```cpp
      // Quantize all const inputs using their values
      for (uint32_t i = 0; i < arity; i++)
      {
        input_node = node->arg(i);
        const_node = dynamic_cast<luci::CircleConst *>(input_node);
        if (const_node != nullptr && !is_quantized(const_node))
          quant_const(const_node, output_type);
      }
```

But, since `Gather` has integer type for `gather->indices()`, I think it would be thrown by following assert.

https://github.com/Samsung/ONE/blob/89f95897ab4864e8bce8b6acdfe4c22a1db9308f/compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp#L303

Reply:
How about adding `CircleGather` to `VerifyQuantizedNodeGranularity`?
Not exactly but maybe such like following example?

```cpp
  bool visit(const luci::CircleGather *node)
  {
    RETURN_FALSE_UNLESS(is_lwq(node));
    RETURN_FALSE_UNLESS(is_cwq_const(node->params()));
    return true;
  }
```

Reply:
@jyoungyun , @llFreetimell , there is `if (const_node != nullptr && !is_quantized(const_node))` for each input.. that will skip for `indices()` as I understand. Did I miss something?
```cxx

bool is_quantized(const CircleNode *node)
{
  return node->quantparam() != nullptr &&
         (node->dtype() == loco::DataType::U8 ||  // activation, weight (uint8 quant)
          node->dtype() == loco::DataType::S16 || // activation, weight (int16 quant)
          node->dtype() == loco::DataType::S32 || // bias (uint8 quant)
          node->dtype() == loco::DataType::S64);  // bias (int16 quant)
}
```

Reply:
I've moved the `case` line that checks only for arg(0)

Reply:
You've moved Greater, not Gather.

Reply:
I thought that if `indices()` is `CircleConst` with `S32`,

- `const_node != nullptr` : true
- `node->quantparam()` != nullptr : false

So I thought that `assert` would throw... T.T
But anyway this is resolved now..! :D

Reply:
> Did I miss something?

At this point, Const input of Gather is not yet quantized (quantparam() == nullptr, so `is_quantized` returns false).

BTW, Sorry for the wrong review. I thought that this PR is the same with #8265 , but it wasn't.

Reply:
😭